lib/debezium/decimal.go

```diff
@@ -90,24 +89,26 @@ func decimalWithNewExponent(decimal *apd.Decimal, newExponent int32) *apd.Decima
 	}
 }
 
-// EncodeDecimal is used to encode a string representation of a number to `org.apache.kafka.connect.data.Decimal`.
-func EncodeDecimal(value string, scale uint16) ([]byte, error) {
-	decimal, _, err := new(apd.Decimal).SetString(value)
-	if err != nil {
-		return nil, fmt.Errorf("unable to use %q as a floating-point number: %w", value, err)
-	}
-
-	targetExponent := -int32(scale) // Negate scale since [Decimal.Exponent] is negative.
-	if decimal.Exponent != targetExponent {
-		decimal = decimalWithNewExponent(decimal, targetExponent)
-	}
-
+// EncodeDecimal is used to encode a [apd.Decimal] to [org.apache.kafka.connect.data.Decimal].
+// The scale of the value (which is the negated exponent of the decimal) is returned as the second argument.
+func EncodeDecimal(decimal *apd.Decimal) ([]byte, int32) {
```

Comment:
I chose to use `int32` for the scale because that's both what `apd.Decimal` uses as well as the wire representation of `org.apache.kafka.connect.data.Decimal`.

Reply:
Make this private?

Reply:
Unless you are planning to have reader call EncodeDecimal and EncodeDecimalWithScale depending on whether we know the scale or not

Reply:
Reader is going to call both functions in different places.

Reply:
Hah your second comment didn't show up for me when I responded!